data_collection/gazette/spiders/sp_americana.py

```diff
@@ -0,0 +1,68 @@
+import dateparser
+from scrapy import Spider
+import datetime
+from dateutil.relativedelta import relativedelta
+from gazette.items import Gazette
+import re
+
+
+class SpAmericanaSpider(Spider):
+    TERRITORY_ID = "3501608"
+    name = "sp_americana"
+    allowed_domains = ["diariooficial.americana.sp.gov.br"]
+    start_date = datetime.date(2018, 9, 1)  # First gazette available
+    start_urls = ["https://diariooficial.americana.sp.gov.br/diario-oficial-edicaoAnterior.php"]
+    extra_editions = "https://diariooficial.americana.sp.gov.br/diario-oficial-edicaoExtra.php"
+
+    locations = {
+        "gazette": "//*[@class='day table-light']/strong/a",
+        "extra_gazette_url": "//*[@class='list-group-item']/a/@href",
+        "extra_gazette_date": "//*[@class='list-group-item']/text()"}
+
+    def parse(self, response):
+
+        param_url = f"?mes={self.start_date.month}&ano={self.start_date.year}"
+        base_url = response.url
+        url = base_url + param_url
+        date = self.start_date
+
+        while date.year < datetime.date.today().year or date.month < datetime.date.today().month:
+            yield response.follow(url, self.parse_gazette)
+            date = date + relativedelta(months=+1)
+            url = base_url + f"?mes={date.month}&ano={date.year}"
+
+        yield response.follow(self.extra_editions, self.extra_parse_gazette)
+
+    def parse_gazette(self, response):
+
+        gazettes = response.xpath(self.locations["gazette"]).extract()
+        for gazette in gazettes:
+            file_url = gazette.split('"')[1]
+            date = gazette.split('"')[7][6:16]
```

Comment:
**suggestion** Extracting data from strings using slices may be dangerous as a simple extra space in the string that can be manually added can break. For example, if they change `Data: 28/08/2021` with `Data:28/08/2021` (a very likely change if they add this information manually), your slice will not work properly. A regex is a bit more safe for this situation.

```
date = gazette.css("::attr(title)").re_first(r"(\d{2}\/\d{2}\/\d{4})")
```

This regex extracts from the string inside `title`, any string that has 2 digits followed by a `/`, then more two digits, another `/` and then 4 digits. So you will end with date equal to `03/08/2021` for example.

Reply:
adjusted!